gms/api/src/main/pegasus/com/linkedin/dashboard/ChartKey.pdl

```diff
@@ -0,0 +1,24 @@
+namespace com.linkedin.dashboard
+
+/**
+ * Key for chart resource
+ */
+record ChartKey {
+  /**
+   * The name of the dashboard tool such as looker, redash etc.
+   */
+  @validate.strlen = {
+    "max" : 20,
+    "min" : 1
+  }
+  tool: string
+
+  /**
+   * Unique id of the chart assigned by the specific dashboard tool
+   */
+  @validate.strlen = {
```

Comment:
SpotHero uses Looker. Looker has different types of entities. For example:
- "Looks", which are just single visualizations.
- "Dashboards", which are collections of visualizations displayed on one page.
- "Boards", which are collections of Looks or Dashboards.

Each has a different URL structure:
- Looks: `${BASE_URI}/looks/${LOOK_ID}`
- Dashboards: `${BASE_URI}/dashboards/${DASHBOARD_ID}`
- Boards: `${BASE_URI}/browse/boards/${BOARD_ID}`

I can forsee some problems with `chartId` and trying to distinguish between look IDs vs dashboard IDs vs board IDs. There are different ways to resolve this problem, but the simplest would be getting rid of the idea of a `chartId` and instead have a `chartUri` and just storing the URI of the visualization tool where the component exists.

Reply:
Thanks for your review @grantatspothero and getting a review from a Looker expert is definitely very helpful (I never used Looker) :)

In the design phase, I identified two entity types - [charts & dashboards](https://github.com/linkedin/datahub/tree/master/docs/rfc/active/1778-dashboards#entities), which are common between different dashboarding tools. Please also review RFC for dashboards.

From your comment, I understand that `board` would also be considered as a separate entity type (at least for Looker). Is my understanding correct? We can add that as an entity but do you think it's a generic concept that can be applied to any other dashboarding tool or is it Looker specific?

Reply:
`Board` is a very Looker-specific concept and totally makes sense to leave out as a first pass 👍

Reply:
> it might make sense to store a chartURI explicitly
How is `chartURI` defined? Is it including the `BASE_URI` in your original definition? URN should have two main characteristics: unique & not changing. I feel like `BASE_URI` might be changing like v1 becomes v2 etc or your looker server's name (or IP) changes. If that happens, you'll have two copies of metadata for the same chart/dashboard.

> constructing a URI from a chartid is non-trivial based on the type of chart
This is where I'm confused at. If we store `LOOK_ID` as chartId in the URN, are you not able construct a chart URI using ${BASE_URI}/looks/${chartId}. What do you mean by chart types? We have a separate URN for dashboards. And, we will use it for Looker dashboards. Only thing we're missing is `boards` at this point.

Reply:
> How is chartURI defined? Is it including the BASE_URI in your original definition? URN should have two main characteristics: unique & not changing. I feel like BASE_URI might be changing like v1 becomes v2 etc or your looker server's name (or IP) changes. If that happens, you'll have two copies of metadata for the same chart/dashboard. I don't think we want that?

Yes I was including the base uri in the chart uri. If datahub does not explicitly store a chartURI then how do you map from a unique chart id back to a URI? It seems important for datahub to surface a link to the chart since a chart id of eg: `522` does not actually mean anything. 

If datahub does not store a chartURI, then it would be on datahub to understand that a `chart` in Looker actually means a `Look` and then template in the chart id into some looker look URI base template. See my below comment for why this can get complicated when considering "looks" vs "explores"

Reply:
We have couple use cases of external links in internal DataHub. The approach we've taken for that is to basically store `BASE_URI` as a global config to generate the external links. But this has the underlying assumption that combination of BASE_URI and stored chart id will be enough to uniquely produce the link for the chart. If not, that means we need to store more metadata for the chart.

After reading your below comment, I started to think of looks & explores as two different customizations of chart (think of chart as the parent class, looks & explores as child classes). But, my understanding could still be limited.

Reply:
I think different chart types with customizable base URIs makes sense. Will allow support different dashboarding tools.

Another consideration is supporting multiple dashboard instances per datahub install. Is this something you are trying to support? For example, SpotHero uses two Looker instances (one large instance for internal use, and one small instance for external use) and the base URIs would need to be different even though both are "Looker" charts.

Reply:
According to our meeting, I'll change the definition of `chartId` as `chartURI`.